cdap-docs/developer-manual/source/metadata/field-lineage.rst

```diff
@@ -0,0 +1,195 @@
+                                                 "address");
+
+Finally ``WriteOperation`` which represents the fields are being written to the CDAP dataset say "passenger"::
+
+    Operation write = new WriteOperation("Write", "Wrote to passenger dataset", EndPoint.of("ns", "passenger"),
+                                         Arrays.asList(InputField.of("Read", "id"),
+                                                       InputField.of("Concat", "fullName"),
+                                                       InputField.of("Normalize", "address")));
+
+Note that ``TransformOperation`` and ``WriteOperation`` both take list of ``InputField`` as a parameter
+representing input fields to the operation. InputField is a pair of origin and field name which identifies the
+input unambiguously. For example: ``Write`` operation above uses the ``address`` field generated by ``Normalize``
+operation and not the one generated by ``Read`` operation.
+
+|
+
+Field operations created above can submitted to the platform in ``initialize()`` method, which is invoked at
+runtime before MapReduce or Spark program is executed::
+
+    @Override
+    public void initialize() throws Exception {
+      MapReduceContext context = getContext();
+      List<Operation> operations = new ArrayList();
+
+      Operation read = new ReadOperation("Read", "Read passenger information", EndPoint.of("ns", "passengerList"),
+                                         "id", "firstName", "lastName", "address");
+      operations.add(read);
+
+      Operation concat = new TransformOperation("Concat", "Concatenated fields",
+                                                Arrays.asList(InputField.of("Read", "firstName"),
+                                                InputField.of("Read", "lastName")), "fullName");
+      operations.add(concat);
+
+      Operation normalize = new TransformOperation("Normalize", "Normalized field",
+                                                   Collections.singletonList(InputField.of("Read", "address")),
+                                                   "address");
+      operations.add(normalize);
+
+      Operation write = new WriteOperation("Write", "Wrote to passenger dataset", EndPoint.of("ns", "passenger"),
+                                           Arrays.asList(InputField.of("Read", "id"),
+                                                         InputField.of("Concat", "fullName"),
+                                                         InputField.of("Normalize", "address")));
+      operations.add(write);
+
+      // Record field operation
+      context.record(operations);
+    }
+
+**Note that all the operations recorded by the CDAP programs must have unique names.**
+
+Field Lineage for CDAP Data Pipelines
+=====================================
+Plugins in CDAP data pipelines can also record the field lineage. Currently plugins of type ``batchsource``,
```

Comment:
Plugins in CDAP data pipelines can also record the field lineage. Currently, plugins of type `batchsource`, `transform`, and `batchsink` are supported. The capability to record lineage is available in the `prepareRun()` method of the plugin by using the context provided to the `prepareRun` method.

Reply:
fixed.